Userland/Libraries/LibWeb/Loader/LoadRequest.h

```diff
@@ -43,7 +43,7 @@ class LoadRequest {
     void start_timer() { m_load_timer.start(); }
     Duration load_time() const { return m_load_timer.elapsed_time(); }
 
-    Optional<Page&>& page() { return m_page; }
+    Optional<JS::NonnullGCPtr<Page>>& page() { return m_page; }
```

Comment:
This one confused me quite a bit when I was looking at making `Page` GC-allocated. What's actually visting this thing / keeping it alive? I didn't see how that works in `ResourceLoader`

Reply:
(And also please use `GCPtr<T>` instead of `Optional<NonnullGCPtr<T>>` :))

Reply:
> What's actually visting this thing / keeping it alive? I didn't see how that works in ResourceLoader

right, I changed it to use JS::Handle for page pointer because otherwise we can't make sure it won't be deallocated.

Reply:
> right, I changed it to use JS::Handle for page pointer because otherwise we can't make sure it won't be deallocated.

ah okay, makes sense - I was struggling to understand the ownership model. Was thinking something higher up in the stack must have been owning this thing when it was previously ref-counted, since this was never taking a ref. But I suppose it was always just a bug, and now that it is GC-allocated LibJSGCVerifier catches that for us